public/app/features/templating/variable_srv.ts

```diff
@@ -303,6 +311,42 @@ export class VariableSrv {
     this.$location.search(params);
   }
 
+  storeUpdated() {
+    const state = store.getState();
+    let path = this.originalUrl;
+    if (path.indexOf('?') >= 0) {
+      path = path.slice(0, this.originalUrl.indexOf('?'));
+    }
+    if (state.location.path === path) {
+      if (state.location.url === this.originalUrl) {
+        return this.$q.when({});
+      }
+
+      this.originalUrl = this.$location.url();
+      const queryParams = state.location.query; //this.$location.search();
+      if (!Object.keys(queryParams).some(k => k.indexOf('var-') === 0)) {
+        return this.$q.when({});
+      }
+      console.log('VAR Changed', queryParams);
+      return this.$q
+        .all(
+          this.variables.map(variable => {
+            return this.processVariable(variable, queryParams);
```

Comment:
we can't really use this function as is, it is only for dashboard init flows (it checks variable refresh options and will update variables (trigger the variable query) even if it has not changed).

Reply:
This was all a copy from #16128 -- I have updated things so that the location change id detected from redux and that broadcasts an event for the variableSrv